GLIF cells

anujanegi opened the pull request:
Things done:

- renamed and refactored [list_all.py](https://github.com/OpenSourceBrain/AllenInstituteNeuroML/blob/master/CellTypesDatabase/models/GLIF/list_all.py) to  [download_glif.py](https://github.com/OpenSourceBrain/AllenInstituteNeuroML/pull/37/files#diff-8c619315cea57da68fdcd9c64df58279358e417f4e0d810a48cf998631877675) as a few allen SDK functions being used were deprecated.
- updated [run_glif.py](https://github.com/OpenSourceBrain/AllenInstituteNeuroML/blob/master/CellTypesDatabase/models/GLIF/run_glif.py) accordingly. Also modified to simulate with the stimulus downloaded from Allen rather than defining it manually. Chose test sweeps to run for.
- gh-actions to test both the scripts above.

The neuron files downloaded from Allen, also have a different format now. So, the parsing needs to be adapted and updated quite a bit according to this. Do you recommend I use OpenCortext for this, or shift to NeuroMLlte?

@pgleeson @sanjayankur31

pgleeson commented:
Thanks @anujanegi. Yes, NeuroMLlite is probably the way to go.

Was it not possible to choose the same model ids as present in https://github.com/OpenSourceBrain/AllenInstituteNeuroML/tree/master/CellTypesDatabase/models/GLIF to download?

anujanegi commented:
Updates and problems:

- For all the existing older models in the repo: the .nml file is no longer valid, and simulating the LEMS file with pynml, throws the same error as in the gh-action. 
- I updated parsing to currently support 1 Leaky Integrate and Fire (LIF) type. The generated .nml is valid, but simulating the LEMS with pynml throws error.
- Added gh-action to recreate the above error.

Unsure how to proceed, as I don't understand the error, the file does exist.
```
pyNeuroML >>> INFO - Executing: (java -Xmx400M  -jar  "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/pyneuroml/lib/jNeuroML-0.12.0-jar-with-dependencies.jar" -validate ".//GLIF_566291893.net.nml" ) in directory: .
pyNeuroML >>> INFO - Command completed. Output: 
  jNeuroML >>   jNeuroML v0.12.0
  jNeuroML >>  Validating: /home/runner/work/AllenInstituteNeuroML/AllenInstituteNeuroML/CellTypesDatabase/models/GLIF/566291893/./GLIF_566291893.net.nml
  jNeuroML >>  Valid against schema and all tests
  jNeuroML >>  No warnings
  jNeuroML >>  
  jNeuroML >>  Validated 1 files: All valid and no warnings
  jNeuroML >>  
  jNeuroML >>  
pyNeuroML >>> INFO - Loading NeuroML2 file: GLIF_566291893.net.nml
pyNeuroML >>> INFO - Including included files (included already: [])
pyNeuroML >>> INFO - Loading NeuroML2 file: GLIF_566291893.net.nml
pyNeuroML >>> INFO - Loading LEMS file: ('./LEMS_GLIF_566291893.xml', <pyneuroml.lems.LEMSSimulation.LEMSSimulation object at 0x7f493dc98370>) and running with jNeuroML
pyNeuroML >>> INFO - Executing: (java -Xmx400M  -Djava.awt.headless=true -jar  "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/pyneuroml/lib/jNeuroML-0.12.0-jar-with-dependencies.jar"  "('./LEMS_GLIF_566291893.xml', <pyneuroml.lems.LEMSSimulation.LEMSSimulation object at 0x7f493dc98370>)"  -nogui -I '') in directory: .
pyNeuroML >>> CRITICAL - *** Problem running command: 
1 Leaky Integrate and Fire (LIF)_Sst-IRES-Cre;Ai14-165865.03.01.01
       Command 'java -Xmx400M  -Djava.awt.headless=true -jar  "/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/pyneuroml/lib/jNeuroML-0.12.0-jar-with-dependencies.jar"  "('./LEMS_GLIF_566291893.xml', <pyneuroml.lems.LEMSSimulation.LEMSSimulation object at 0x7f493dc98370>)"  -nogui -I ''' returned non-zero exit status 1.
<Lems>
pyNeuroML >>> CRITICAL -  jNeuroML >>   jNeuroML v0.12.0

 jNeuroML >>  File does not exist: ('./LEMS_GLIF_566291893.xml', <pyneuroml.lems.LEMSSimulation.LEMSSimulation object at 0x7f493dc98370>)
```

Am I missing something obvious here? @pgleeson

pgleeson commented:
Looks good!
